python/lsst/ts/externalscripts/auxtel/latiss_base_align.py

```diff
@@ -880,3 +894,37 @@ async def run(self) -> None:
         await self.assert_feasibility()
 
         await self.arun(True)
+
+    async def cleanup(self):
+        if self.state.state != ScriptState.ENDING:
+            # abnormal termination
+            if self.iterations_started and self.iterations_executed != self.max_iter:
+                self.log.warning(
+                    f"Terminating with state={self.state.state}: stop iterations. "
+                    f"Iterations started: {self.iterations_started}. "
+                )
+                try:
+                    self.log.warning(
+                        f"Looping failed at iter {self.iterations_executed} from a total of "
+                        f"{self.max_iter}. Returning hexapod to its initial position.\n"
+                    )
+
+                    self.rem.ataos.evt_detailedState.flush()
```

Comment:
I think here you will have to do something else. Because the loop could be interrupted while it was taking the intra/extra focal image. I think in this situation what you want to do is get the offsets before the script started executing, then get the offsets at this point and apply the offset to restore them to the initial position.

I think you could just put this procedure in a method and then call it here.